zerver/lib/push_notifications.py

```diff
@@ -736,15 +749,74 @@ def push_notifications_configured() -> bool:
 
 
 def initialize_push_notifications() -> None:
+    """Called during startup of the push notifications worker to check
+    whether we expect mobile push notifications to work on this server
+    and update state accordingly.
+    """
+
+    if sends_notifications_directly():
+        # This server sends push notifications directly. Make sure we
+        # are set to report to clients that push notifications are
+        # enabled.
+        for realm in Realm.objects.filter(push_notifications_enabled=False):
+            do_set_realm_property(realm, "push_notifications_enabled", True, acting_user=None)
+            if realm.push_notifications_enabled_end_timestamp is not None:
+                do_set_push_notifications_enabled_end_timestamp(realm, None, acting_user=None)
```

Comment:
I added these blocks to clear `push_notifications_enabled_end_timestamp` if set. I expect this code to mostly not do anything, but in theory if a server had a previous value of `push_notifications_enabled_end_timestamp`, then it should be cleared in this case.